src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,52 @@
+import { useContext, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { ArticleContext } from "../App";
+import baseApi from "../../api";
+
+export default function SingleArticle() {
+  const { articleId } = useParams();
+  const { articles, isLoading } = useContext(ArticleContext);
+  const [article, setArticle] = useState({});
+
+  useEffect(() => {
+    baseApi
+      .get(`/articles/${articleId}`)
+      .then((res) => {
+        setArticle(res.data.article);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, [articleId, articles]);
```

Comment:
why are we using articles as a dependency here ? Do we not want to fetch the article above when the page loads , so do we need to put anything in this dependency array